e3nn/rsh.py

```diff
@@ -0,0 +1,167 @@
+# pylint: disable=not-callable, no-member, invalid-name, line-too-long, unexpected-keyword-arg, too-many-lines
+"""
+Real Spherical Harmonics equivariant with respect to o3.rot and o3.irr_repr
+"""
+import math
+
+import torch
+from scipy import special
+
+from e3nn.util.default_dtype import torch_default_dtype
+
+
+def spherical_harmonics_expand_matrix(lmax):
+    """
+    :return: tensor [l, m, l * m]
+    """
+    m = torch.zeros(lmax + 1, 2 * lmax + 1, sum(2 * l + 1 for l in range(lmax + 1)))
+    i = 0
+    for l in range(lmax + 1):
+        m[l, lmax - l: lmax + l + 1, i:i + 2 * l + 1] = torch.eye(2 * l + 1)
+        i += 2 * l + 1
+    return m
```

Comment:
Should we mention where this is used so a new person would understand what it is for better? I.e. in the KernelLinear

Reply:
Ok!